### DISCUSSION TITLE
區塊拖曳實做問題(drag event)

### DISCUSSION THREAD

**Discussion by EshauHuang** [Lidemy/forum#217] [前端相關問題] [answered]
遇到了如題的問題，感謝各位的協助🙇

附上完整 [程式碼](https://codesandbox.io/s/great-nash-2qfsm?file=/src/App.js)，目前可以做到讓元素拖曳到上下的區塊，但當元素拖曳到「非原本的區塊」時，會沒辦法觸發 `onDragEnd`，因此無法正常執行原本想在觸發這個事件後，將元素顏色從灰色改回白色的功能。

目前猜測原因是在 `handleDragOver` 這邊改變元素的方式，當傳入的 type 改變，則會改變「目前拖曳的元素」的 type 的值(依照 dragging 判斷)，這樣重新 render 後就可以依照 type 重新排列元素，但可能就是這個「重新排列元素」導致我的 `onDragEnd` 沒辦法作用，因為當元素沒有移動到其它區塊就放開時， `onDragEnd` 還是可以作用，所以我才會猜測是因為這個原因，但其實沒有很確定。

**Comment by aszx87410** [comment, marked as answer]
看起來好像是一個 react 的 bug：https://stackoverflow.com/questions/24537000/react-js-events-not-firing-for-the-last-rendered-element
但我不太確定就是了，我看了半小時看不出什麼東西，但我也猜測是 rerender 造成的問題
解不掉的話可以改用其他套件試試看：https://github.com/atlassian/react-beautiful-dnd

**Reply by EshauHuang** [reply]
感謝老師的解答，來試看看這個套件！
